contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -287,7 +282,7 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         // Check the message has not been delivered before by checking that there is no relayer reward
         // address stored for it already.
         require(
-            relayerRewardAddresses[warpMessage.sourceChainID][
+            _relayerRewardAddresses[warpMessage.sourceChainID][
```

Comment:
I think instead of checking `_relayerRewardAddresses` directly here, we should make an internal `_messageReceived` and use that check here to make it more clear.

Reply:
That wasn't an item called out in the audit, so I think it's out of scope for this PR.

Reply:
Gotcha, I personally am not super inclined to make the change given it increases overall contract size and gas costs and the check is properly described in the comment directly above here. If others feel strongly though I'm open to it.

Reply:
When you say `gas costs` do you mean the increase in gas for deployment, or are you expecting an increase in gas for wrappign the check now in a function call?

Either way think the gas cost is minor in favor of more readable code so would be my vote.

Reply:
Fair enough, yeah I checked the `forge test --gas-report` before and after making those change, and it was a very minor increase. 

I added an internal `_messageReceived` now.